### PR TITLE
WT-12459 Remove session rename tests and testing frameworks

### PR DESCRIPTION
**Pull request by ruby-oujo**

### CODE REVIEW

File: test/suite/test_tiered13.py [hunk 1]
```diff
@@ -140,19 +138,3 @@ def test_tiered13(self):
         # Try to import via the file:uri with file metadata.
         self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
             lambda: self.session.create(self.file2uri, import_meta), invalid)
-
-        # Try to import via a renamed object. If we don't send in metadata,
```
Comment (ruby-oujo) [inline test/suite/test_tiered13.py:144]: This test I am unsure about; we are renaming the file (it says object here though) without using the session rename API. However, if we are removing this feature then is this testing redundant?

File: test/suite/test_bug006.py [hunk 2]
```diff
@@ -54,21 +54,17 @@ def test_bug006(self):
...
-            wiredtiger.WiredTigerError,
-            lambda: self.session.rename(uri, self.uri + "new", None))
         self.assertRaises(
             wiredtiger.WiredTigerError, lambda: self.session.salvage(uri, None))
         self.assertRaises(
             wiredtiger.WiredTigerError, lambda: self.session.upgrade(uri, None))
         self.assertRaises(
             wiredtiger.WiredTigerError, lambda: self.session.verify(uri, None))
 
+        self.session.checkpoint()
```
Comment (korteland) [inline test/suite/test_bug006.py:64]: Why was this needed?

File: test/suite/test_lsm02.py [hunk 3]
```diff
@@ -59,22 +59,20 @@ def test_lsm_tombstone(self):
         self.add_key(self.uri, 'k3', v)
         self.verify_key_exists(self.uri, 'k3', v)
 
-    def test_lsm_rename01(self):
+    def test_lsm_create01(self):
```
Comment (korteland) [inline test/suite/test_lsm02.py:62]: I think this whole function can be dropped - the test is now just adding a key then verifying it exists.

File: test/suite/test_lsm02.py [hunk 4]
```diff
@@ -59,22 +59,20 @@ def test_lsm_tombstone(self):
...
         self.session.create(self.uri, 'key_format=S,value_format=S')
         self.add_key(self.uri, 'a', 'a')
-        self.renameUntilSuccess(self.session, self.uri, self.uri + 'renamed')
-        self.verify_key_exists(self.uri + 'renamed', 'a', 'a')
+        self.verify_key_exists(self.uri, 'a', 'a')
 
-    def test_lsm_rename02(self):
+    def test_lsm02_create02(self):
```
Comment (korteland) [inline test/suite/test_lsm02.py:67]: Again, this function can (I think) also go. It's just two creations, insertions, then assertions that the key exists.

File: test/suite/test_schema08.py [hunk 5]
```diff
@@ -158,8 +154,8 @@ def run_recovery(self, uri, suburi):
     # Test that creating and dropping tables does not write individual
     # log records.
     def test_schema08_create(self):
-        if self.is_tiered_scenario() and (self.uri == 'lsm:' or self.uri == 'file:' or self.schema_ops == 'rename'):
-            self.skipTest('Tiered storage does not support LSM or file URIs, and also does not support the rename operation.')
+        if self.is_tiered_scenario() and (self.uri == 'lsm:' or self.uri == 'file:'):
+            self.skipTest('Tiered storage does not support LSM and file URIs.')
```
Comment (korteland) [inline test/suite/test_schema08.py:158]: I think the original `or` was correct. But you and I have some shared history with grammar arguments...

File: test/suite/wthooks.py [hunk 6]
```diff
@@ -56,14 +56,12 @@ def tty(message):
 # Here are some useful terms to know, with some commentary for each.
 #
 #   API functions
-#      potentially any WiredTiger API functions that a hook creator wishes to modify (like
-#      Session.rename).  In Python most everything is an object.  Of course an instance of
-#      "Session" is an object, but also the "Session" class itself is an object.  The Session.rename
-#      function is also an object (of a certain form that can be called).  Also in Python,
+#      potentially any WiredTiger API functions that a hook creator wishes to modify. In Python most everything is an object.  Of course an instance of
```
Comment (korteland) [inline test/suite/wthooks.py:59]: Why not change this example to `Session.remove` as well?

File: test/suite/wttest.py [hunk 7]
```diff
@@ -717,8 +717,8 @@ def timestamp_str(self, t):
 
     # Some tests do table drops as a means to perform some test repeatedly in a loop.
     # These tests require that a name be completely removed before the next iteration
-    # can begin.  However, tiered storage does not always provide a way to remove or
-    # rename objects that have been stored to the cloud, as doing that is not the normal
+    # can begin.  However, tiered storage does not always provide a way to remove
```
Comment (korteland) [inline test/suite/wttest.py:720]: I think the word "objects" was accidentally also consumed here.

File: test/suite/test_tiered13.py (hunk 1 above)
Comment (korteland) [reply test/suite/test_tiered13.py:144]: Looking at the code above, I think this rename was done via copying the file. So this isn't a `session.rename` usage, and should probably stay. One way to check would be to make `session.rename` always fail in WiredTiger, and see if this test still passes.

Comment (ruby-oujo) [reply test/suite/test_tiered13.py:144]: Hi @korteland, I agree, I do not think it is using the session rename API (as previously mentioned) - I am more so concerned with whether or not we want to continue testing for a scenario that we aren't going to be using in the future - thanks!

File: test/suite/wthooks.py (hunk 6 above)
Comment (ruby-oujo) [reply test/suite/wthooks.py:59]: I thought it was a bit repetitive but I can add it back.

File: test/suite/test_bug006.py (hunk 2 above)
Comment (ruby-oujo) [reply test/suite/test_bug006.py:64]: Without a checkpoint, the session salvage call returns an error "resource busy". This may be because we need to clear dirty data. I can alternatively change session.salvage to use the function` salvageUntilSuccess()` - which loops  calling salvage, checking if the error is EBUSY and checkpointing.

File: test/suite/test_lsm02.py (hunk 4 above)
Comment (ruby-oujo) [reply test/suite/test_lsm02.py]: [4651212](https://github.com/wiredtiger/wiredtiger/pull/10282/commits/4651212791deac3577df5658fab5aa3c0f6f45c4)

File: test/suite/test_schema08.py (hunk 5 above)
Comment (ruby-oujo) [reply test/suite/test_schema08.py]: Haha I am not fussed about this, just thought it read better - it means the same thing. Updated here [4651212](https://github.com/wiredtiger/wiredtiger/pull/10282/commits/4651212791deac3577df5658fab5aa3c0f6f45c4)

File: test/suite/wthooks.py (hunk 6 above)
Comment (ruby-oujo) [reply test/suite/wthooks.py]: [4651212](https://github.com/wiredtiger/wiredtiger/pull/10282/commits/4651212791deac3577df5658fab5aa3c0f6f45c4)

File: test/suite/wttest.py (hunk 7 above)
Comment (ruby-oujo) [reply test/suite/wttest.py]: good spotting, [4651212](https://github.com/wiredtiger/wiredtiger/pull/10282/commits/4651212791deac3577df5658fab5aa3c0f6f45c4)

File: test/suite/test_bug006.py (hunk 2 above)
Comment (korteland) [reply test/suite/test_bug006.py:64]: Thanks for the explanation! I prefer the sound of `salvageUntilSuccess`, since it appears to map to the original code a little better (presumably there was a reason we called `renameUntilSuccess` - possibly this exact issue).

Comment (ruby-oujo) [reply test/suite/test_bug006.py]: Yup, I figured we needed checkpoint from renameUntilSuccess function - updated here [362d25d](https://github.com/wiredtiger/wiredtiger/pull/10282/commits/362d25dbd979b7aa8202bfd8a11eed3b42e3f2ba)

File: test/suite/test_tiered13.py (hunk 1 above)
Comment (korteland) [reply test/suite/test_tiered13.py:144]: Sorry for the delay getting back to you. I think this test should remain. I don't think the rename is the core of the test here - rather, for tiered storage, it appears to be valid to have "table:xyz-0002.wt" present without the metadata knowing about it, and if that happens I can easily see us needing to be able to import it.

I'm not an expert though - I puzzled this out from staring at the code, so if you want to pull in someone that knows more about tiered storage, that would make sense.

Comment (ruby-oujo) [reply test/suite/test_tiered13.py:144]: I am happy to put the test back - most of the test has been disabled and is waiting on another ticket WT-8664 which is unrelated to the rename API - so I think it can be revisited then.

Comment (korteland) [reply test/suite/test_tiered13.py:144]: Sounds like a plan :-) Thanks for being so patient with this!

Comment (ruby-oujo) [reply test/suite/test_tiered13.py]: [365a0c5](https://github.com/wiredtiger/wiredtiger/pull/10282/commits/365a0c5a173cd414c9dfe504c585362aa8464a08)

**Review by tod-johnson-mongodb — APPROVED** [review]
LGTM
